src/main/java/uk/gov/hmcts/ccd/CachingConfiguration.java

```diff
@@ -22,26 +25,34 @@ public Config hazelCastConfig() {
         NetworkConfig networkConfig = config.setInstanceName("hazelcast-instance-ccd").getNetworkConfig();
         networkConfig.getJoin().getMulticastConfig().setEnabled(false);
         networkConfig.getJoin().getTcpIpConfig().setEnabled(false);
-        configCaches(applicationParams.getDefinitionCacheTTLSecs(), config);
+        configCaches(applicationParams.getDefinitionCacheMaxIdleSecs(), applicationParams.getDefinitionCacheTTLSecs(), config);
         return config;
     }
 
-    private void configCaches(int definitionCacheTTL, Config config) {
-        config.addMapConfig(newMapConfig("caseTypeDefinitionsCache", definitionCacheTTL));
-        config.addMapConfig(newMapConfig("workBasketResultCache", definitionCacheTTL));
-        config.addMapConfig(newMapConfig("searchResultCache", definitionCacheTTL));
-        config.addMapConfig(newMapConfig("searchInputDefinitionCache", definitionCacheTTL));
-        config.addMapConfig(newMapConfig("workbasketInputDefinitionCache", definitionCacheTTL));
-        config.addMapConfig(newMapConfig("caseTabCollectionCache", definitionCacheTTL));
-        config.addMapConfig(newMapConfig("wizardPageCollectionCache", definitionCacheTTL));
-        config.addMapConfig(newMapConfig("userRolesCache", definitionCacheTTL));
+    private void configCaches(int definitionCacheMaxIdle, int definitionCacheTTL, Config config) {
+        config.addMapConfig(newMapConfig("caseTypeDefinitionsCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("workBasketResultCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("searchResultCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("searchInputDefinitionCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("workbasketInputDefinitionCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("caseTabCollectionCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("wizardPageCollectionCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("userRolesCache", definitionCacheMaxIdle));
+        config.addMapConfig(newMapConfig("caseTypeDefinitionLatestVersionCache", definitionCacheMaxIdle, definitionCacheTTL));
```

Comment:
perhaps would be better to have in this case the property called `getVersionCacheTtl`? We can leave also `definitionCacheTTL`. We could have both but definitely we need to be able to control the ttl for the getVersion separately from all the other TTLs

Reply:
Makes sense. Refactored as agreed that the latest version ttl is intended to apply to a few items. Please check.